### ISSUE TITLE
Freedoom1 Phase 1 issues and bugs found v 0.12.1

### ISSUE THREAD

**Issue by hawkwind3** [freedoom/freedoom#698]
e1m7 ... 

**_SERIOUS ... Sector 95 should not do damage. Uses STEP1 for the floor.
Was previously reported by YeldamDev. I just mention the sector concerned._**

e2m1 ... 

1. **IMPORTANT ... Texture on line 378 bleeds into the ceiling in software.**
2. Secret sector 139 only has a backpack on hard skill level.

e2m2 ... 

1. Lines 805 and 880 need to be lower unpegged.
2. No texture clue that sector 11 is a key door.
3. Wrong ceiling flat sector 254 - should be ceil5_1.

e2m3 ... 

Sector 355's floor could be lowered a bit to show all of T300

e2m4 ...

1. Line 722 could be flagged secret.
2. Sector 178 should do damage.
3. Texture on line 1388 bleeds slightly into the floor in software.
4. **IMPORTANT ... Line 1595 has Boom type 85. Should be Doom type 48.**
5. Sectors 273 and 274 would look better with MFLR8_1 for the floor.
6. Line 1617,846,845, 821 could have been flagged secret.

e2m5 ... 

Lines 110, 890, 1255, 2697, 2700, 2699,  could be flagged secret.

e2m6 ... 

1. Lines 345, 235 could be flagged secret.
2. Can walk straight through T115. Should have used blocking type 53.
3. Line 97 needs a Y offset.

e2m8 ... 

End text mentions brutes, plural, but there is only one on easy skill.
Suggest having two "cyberdemons", "invulnerability sphere" and "BFG" on easy skill. 

e2m9 ... 

1. Lines 103, 111, 95, 66, 1691, 3358, 1524, 1514, 3006, 1611 could be flagged secret.
2. Note Ceiling of sectors 333, 335, 334 I think would look better with flat19.

e3m2 ... 

1. Lines 536, 538 could use a Y offset.
2. Lines 553, 561 could be flagged secret.

e3m3 ... 

1. BFALL1 on line 294 etc., and line 321. Maybe dbrain1 for now or a new lava texture in the future.
2. Lines 595, 594, 592 could be flagged secret.
3. Lines 593, 610, 451,615 could use a Y alignment.

e3m4 ... 

1. **IMPORTANT ... Line 2496 needs to be impassible. See line 922. And SP_FACE2 could use an X alignment.**
2. Lines 1479, 1484, 2127, 1932, 1915, 1892, and 3740 etc. could be flagged secret.
3. **IMPORTANT ... Texture on line 524 bleeds into the floor in software.**
4. **_SERIOUS ... Depending on skill level, monsters stuck in each other at sector 392_**.

e3m5 ... 

1. Aashitty at the start could use an x offset.
2. Lines 1624, 21, 364l, 1098, 1103, 1234, 1038 etc., 962 etc. could be flagged secret.
3. Sector 260 would look better with floor7_2 and no damage.
4. Remove X and Y offset on line 152.

e3m6 ... 

1. Line 736 could be flagged secret.
2. Lava adjacent sector 120 does not do damage.
3. Linedef 944 was probably meant to have the texture METAL. See line 946.
4. Monster at sector 171 is a "sitting duck" before crossing line 566. DBRAIN1 would also look better on the lines of sector 171.
Or new lavafall texture when available.

e3m7 ... 

1. Lines 668, 1719, 581, 2273 could have been flagged lower unpegged.
2. **IMPORTANT ... remove mid texture from line 1740.**

e4m1 ... 

1. Lines 597, 600, 601, 682, 688, 800, 805, 1445, 966 could be flagged secret.
2. Was sector 195 supposed to have a ceiling height of 88 ?

e4m2 ... 

Line 1344 could be flagged secret.

e4m9 ... 

Line 2661 could be flagged secret.

e4m3 ... 

1. Lines 801, 1753, 1731, 1735,690, 700, 683, 1361, 1400, 1254, 1269 could be flagged secret.
2. Two main areas of NUKAGE that do different damage.
3. Block sound also needs to be on lines 688, 698 and 681

e4m4 ... 

Line 946 could be flagged secret.

e4m6 ... 

1. Lines 365, 367 need a Y alignment of 61. See line 369.
2. Lines 521 etc., and 501 etc, could use the secret flag.
3. Maybe make line 2343 impassible too. See line 2339.

e4m7 ... 

Line 2524, 1619, 2853/2854 etc. could use the secret flag.

e4m8 ... 

1. Line 1910, 1969, 1974, 1980 could be flagged secret.

2. No "BFG" on any e4 levels. Needs one for e4m8 ! Invulnerability and cell charges.

**Comment by perryprog** [comment]
Right—originally I was going to make some PRs for issues that I've found from when I've played, but since you've made this awesome list (and #694!), I'm going to add them here so it can be all done in one big update.

I'd also like to say that I personally find E4M8 to be pretty balanced. On my casual playthroughs, one of which was UV, I had no issues. I personally don't mind the lack of a BFG, but I had assumed it was intentional. I personally enjoy the aspect that forces you to strafe quickly around the map in order to find the switches to spawn in more techno spiders, and get more infighting, all before you run out of ammo.

---

(Note: I am omitting some issues that I assume are fine, like slope specials in E2M1. As far as I know, they simply have no effect on Vanilla, as opposed to a crash. I also am ignoring a lot of missing textures that are essentially not visible.)

- E1M6
  - Line 2701 has an unused tag for a door close special
- E2M1
  - There is a deathmatch start that has some linedef specials with a nonexistent sector tag. It's 18, and should be 17. Starts at line 410.
- E3M6
  - Line 99 is a start crusher special with tag 15, which no sectors have.
- E3M8
  - **Important:** Line 874 is a generalized special! This causes it's associated secret to be inaccessible in any boom-incompatible port.
  - Line 448 is a stop crusher special with a never-used tag
- E4M1
  -  Line 489 has an unused tag for a door open special
- E4M6
  - **Important:** Line 1465 can crash vanilla/chocolate doom. I'm not totally sure why, but it causes a sector to read a type from a garbage value. The error is `P_PlayerInSpecialSector: unknown special [garbage]`. Two helpful resources: [this](https://tcrf.net/Doom_(PC,_1993)/Revisional_Differences#E3M9:_Warrens) and [this](https://www.doomworld.com/forum/topic/63814-p_playerinspecialsector-unknown-special-258-advice-requested/).
- E4M7
  - There's a teleporter for multiplayer that unintuitively does nothing in single player (which it's not needed for, it's just useful in multiplayer).

---

That's all for now. I have a list for phase two as well, but it's getting late, so I'll have to work on that tomorrow.

**Comment by hawkwind3** [comment]
Hi,

Looks like you've caught my "bug". Pardon the pun.

Well done.

Just let me know when you are finished and then you can add your list
yourself or I can add it to mine, and I'll credit you.

________________________________

**Comment by hawkwind3** [comment]
I have fixed E4M6. Will not crash choco doom now.
[freedoom1_E4M6 _ choco fix.zip](https://github.com/freedoom/freedoom/files/5078330/freedoom1_E4M6._.choco.fix.zip)
While at it I addressed the other issues I mentioned.

**Comment by hawkwind3** [comment]
@perryprog ... could you recheck those lines for e3m8. They don't make sense with DB2.

**Comment by perryprog** [comment]
That was meant to be E3M7, sorry. I've fixed my comment.

**Comment by perryprog** [comment]
@hawkwind3 That fixes E4M6. Out of curiosity, what did you change? I can't seem to tell as there's no way to diff binary WADs sadly.

**Comment by Proluiz** [comment]
i can take care of episode 2, with given time.

**Comment by hawkwind3** [comment]
@**Proluiz** ... go for it. I suggest working along with **perryprog** as he will be adding to the episode 2 list.

**Comment by hawkwind3** [comment]
> 
> 
> @hawkwind3 That fixes E4M6 for me, nice job. Out of curiosity, what did you change? I can't seem to tell as there's no way to diff binary WADs sadly.

Your second link I think reminded me of the solution. All the lines bounding sector 502 are numbered. I had to copy/paste various lines at times to make new linedef 1460 the lowest numbered linedef.
> 
> 
> That was meant to be E3M7, sorry. I've fixed my comment.

Easily fixed. Just change the line type on linedef 874 to 105.
All the other stuff without tags are benign. Just remove the action/type if you wish.

**Comment by Proluiz** [comment]
e2m1 actualy has some deathmatch room/closets

**Comment by hawkwind3** [comment]
Yes ... and ... ? I'm not sure what you are trying to say.

I cannot see an issue with sectors 69, 95 and 99.

Do you mean something else ?

________________________________

**Comment by perryprog** [comment]
@hawkwind3 I'm so sorry; I meant to say it started at linedef 417. It's a ring of W1 Door Close linedefs, but the tag they use (18) is never referenced. I'll update my comment with the correct number.

**Comment by Proluiz** [comment]
one thing i want to comment on missing textures:

I dont like to inserting textures in lines that is impossible to see, for example in walls of 0 height sectors, there's just "buried" and outside player view so that's not necessary to have a texture.

**Comment by hawkwind3** [comment]
OK

________________________________

**Comment by hawkwind3** [comment]
I have no problem with that.

________________________________

**Comment by hawkwind3** [comment]
> 
> 
> @hawkwind3 I'm so sorry; I meant to say it started at linedef 417. It's a ring of W1 Door Close linedefs, but the tag they use (18) is never referenced. I'll update my comment with the correct number.

Now I see what you mean.  It's benign. If you wish you could just remove those linedef actions and tag 18.

**Comment by Proluiz** [comment]
regarding about e2m5 why tag line 3605 as secret?
i see no real logic on it...

**Comment by Proluiz** [comment]
e2m6 and e2m8 i'm skiping, personal reason...
And @perryprog have you did any progress in the issues you mentinoed so we can merge our wads, otherwise i'll contune following the list an progressing on hawkwind3 episode 3 and 4 issues them move up to your list as well, awaiting for a awsner.

**Comment by perryprog** [comment]
@Proluiz Oh, I'm sorry. I was originally planning on waiting for a comment from a maintainer before moving forward, but I then forgot about this issue and became busy with school. I'm not sure yet if I'll have time to fix the things I mentioned—I may have a better idea soon (e.g., this week).

One thing I'm a little worried about is the ease of merge conflicts occurring as Doom-style maps are binary. This could potentially make asynchronous work on the same map a pain.

(Side note: A UDMF to Doom-style maps would be _really_ cool. No idea if it would be viable, but it sure would be neat.)

**Comment by BailShamber** [comment]
Add as well to E1M7 that the player can get stuck in the rising pillars with ammo, if the player has sufficient health and armor.
![Screenshot from 2020-09-08 01-57-51](https://user-images.githubusercontent.com/49800926/92424872-d71a7080-f185-11ea-9d6c-415e3cd96c61.png)

![Screenshot from 2020-09-08 02-13-53](https://user-images.githubusercontent.com/49800926/92424897-eac5d700-f185-11ea-9400-dea7c5cc7ba4.png)


![Screenshot from 2020-09-08 02-14-10](https://user-images.githubusercontent.com/49800926/92424919-f6b19900-f185-11ea-9d2f-133a82ab7552.png)

**Comment by hawkwind3** [comment]
You're right. Updated my original post accordingly.

________________________________

**Comment by Proluiz** [comment]
ok leave this whole issue on my hands
i'll take care of all them and soon release a PR.

**Comment by Proluiz** [comment]
i'll be skipping episode 4 for a while, and start doing what is in perryprog list.

**Comment by Proluiz** [comment]
ok all fixed prior to episode 4 was fixed, and i'm delaying the episode 4 for depth investigation of these reports, i'm not familiar with this episode yet.
all will be send in a big PR.
i also want to suggest to you both if possible do a single list of episode 4 bugs so i can have a easy time looking at them.
Note: Certain map fixes was skipped because they have a high chance of being edited/replaced in soon, like e3m6, a new map is under dev by harbinger.

**Comment by Proluiz** [comment]
commit https://github.com/freedoom/freedoom/commit/7cf74962b51ba4ed2639e322745376e12ad84cf9 fix all issues prior to episode 4 for now that's it.

**Comment by hawkwind3** [comment]
@Proluiz 
Good work fixing most of E1 to E3 issues.

I noticed that E2M2 and E3M6 issues were not mentioned in your fixes.

**Comment by Proluiz** [comment]
they were no edited at all:
E2M2 is being replaced by a improved version that i'm close to finishing so no reason fising something that's about to be edited, and same things for e3m6 but by zero the red

**Comment by hawkwind3** [comment]
OK. Fair enough.

________________________________

**Comment by Xindage** [comment]
well i decided myself to start editing episode 4 and them follow up with a very detailed changelog of what was changed, Since this map pack was donated to freedoom it wont mean so that cannot be edited,

So if any test tell me that something was intended to be or not, we'll have a fixed version of the episode.

**Comment by andh0g** [comment]
This issue is still open.  Is it meant to receive new stuff discovered or should I open a new issue?  Everything refers to the latest version in the repository.  Sorry if the text is a bit lengthy.

I'm just done with playing through Phase 1 and here is my first contribution to the project by giving some feedback.  First of all, thanks to all contributors for a lot of great levels!

The first two things that are clearly bugs.

E3M6, sector 62, at about x: -980, y: 1080, is a pit without exit and no radioactivity to kill you.  I suggest either to make the pit radioactive or to provide a way out.

E3M7: There is a Trilobite outside of everything at x: 10880, y: 24608.  It can not be killed since it is outside of every room in the level and effectively prevents you from getting 100% kill.

E4M6: The switch at about x: 526 y: 753, linedef 1074, on the back of a pillar is hard to operate since it is blocked by a column.  I understand that the point is that it shouldn't be too easy to spot it but could something be done to make it easier to reach the switch?  Maybe widen the switch somewhat or put the column a little bit off sideways?

E4M8: There is an unreachable construct at x: 2430 y: -960, sector 127, 169 and 196, that shows up on the Tactical Survey Map.  I see no purpose with that and I find it confusing when looking for secrets.  I suggest that either remove this construct or mark it as non-visible on the map if it has some technical purpose.

In several places there there are some 1%-health or 1%-armour bonuses stacked together making the appearance that there exist, e.g., 5% bonuses.  I think this is bad, the documentation clearly states they should be 1%.  I suggest spreading them out so it becomes clear that there are several of them.  I have found them at the following locations: E1M5 x: -1360 y: -904, E3M5 x: -1824 y: -2432 and in E3M4 all over the place (30 places!).

I find it hard to know what floors give damage.  The very same floor texture that gives damage on one level, doesn't on another.  I have no particular example here.  Is this a deliberate choice or just a consequence of many contributors?  Is there a style guide for Freedoom levels that says anything about this?

Exits

In my opinion it should be clear when operating a switch whether it ends the level or not.  I really hate it when I accidentally end a level while still having some enemies left or some secrets still to be found.  In most levels this is clear, for instance by use of a well placed exit sign or a for that episode consistent way of how the exit looks.

Episode 1 seems to be rather consistent with use of a exit sign and the special exit door.  However, on level E1M2, the exit sign is very hard to spot since it is very dark.  I suggest to move the sign into the wall where it can be lit up better. There is no exit sign at all in E1M4 and E1M6.  Should be easy to add one.

Episode 2 seem to start with using the special portal with the Fireblue texture but the consistency is lost after the first four levels.  I recognise that it with so many contributors can be hard to keep a consistent look and feel within each episode, especially if levels are moved between episodes.

E2M7 has no sign or other marking of the exit as far as I can see.  I have no suggestion for how to change that and I have not made up my mind if I think it is OK as it is.

E2M9 have exit signs in at least three places.  Even though they are related to the exit, I find them confusing. Should be removed except for the actual exit.  Or could a new texture with a different looking exit sign be introduced for this?

Episode 3 introduces the teleporter exit with the special texture Gate4.  However, this texture is used so frequently for other teleporters and also non-teleporters so it does not signal end of level.

E3M5 introduces the hole-in-the-floor exit.  This is well known for old Doom players but someone new to the game will not recognise this.  I think it would have been great to use consistently as exit for a complete episode though.  Just to confuse things, E3M2 have already introduced the hole in the floor as a normal teleporter.  I suggest changing the teleporter hole in E3M2 to a normal teleporter, perhaps out of view from the windows into the room.

Episode 4 seems to go back to the exit door from episode 1, usually together with exit sign.  Only E4M9 is unmarked and should perhaps have an exit sign somewhere.  E4M6 looks more like a button to operate the elevator next to it.  I suggest either to remove the fake elevator or make the button operate the elevator and let it lead to a new room with the exit button in it.

Secrets

Secrets are clearly subject to opinion!  My opinion is that secrets are fun but should be discoverable.  You should not need to walk around the complete level trying to operate all walls, at least not if the level is fairly large.  The secrets should, of course, not be too easy either.

In E3M4 there is a secret at x: 2350 y: -5400, sector 363, where I still see no clue that would have drawn my attention to this one.

In E3M5 there is a hidden switch at x: -3920 y: -1870 on the back of a pillar that opens the door just behind the start.  The texture is differently aligned compared to the other sides of the pillar but it is too subtle, in my opinion, for a level of this size.  Maybe remove the "cover plate" over the switch or change the texture to something clearly different (it is on the back of the pillar!).


Discoverable can also mean that you see a space, perhaps with some power ups to make it clear that you should be able to get there, but without an obvious path to the place.  The location narrows it down where to look for a hidden door or switch.

In many levels there are unreachable areas whose only apparent purpose is to provide a decor background, perhaps with some monsters.  These areas confuse me when looking for secrets.  This might, of course, be a deliberate design decision.  If not, it might be substantial amount of work to change it.

One example of such an area is in E1M7 around x: 1600 y:700.  There are no power ups there but there are ammunition from killed monsters.  Especially the doors make it look like an area I should be able to get to.  In this case, it should be possible to make the area reachable without any major redesign of the surrounding area.


A secret that is discoverable but very hard to reach is in E4M7 at x: -1320 y: -140, sector 320.  You have to trigger the door in a different place, have perfect timing with the elevator and use diagonal running to reach it before the door closes.  Maybe it matters that I'm a keyboard player (is there a trick for mouse players to run even faster?).  I think it is too hard to reach.  I got the impression that there was some other way to reach this area but haven't found any.


Another thing I dislike in connection to secrets are areas reachable only once or exit areas you cannot return from.  It makes it much harder to find that last secret.  It is also annoying when you reach the exit and see that you haven't yet killed all monsters and cannot go back to finish them off.  An argument against what I'm saying is that it's part of the game to learn the level and replay it; It's no different from an ambush that kills you.

Such areas exist in E2M6 where you get surprisingly teleported to it just before the exit.  A place with an exit area you cannot return from is in E3M2 after jumping through the hole-in-the-floor teleporter.  Another place is in E3M7 after the jump behind the yellow door.

A place where it is OK with a visit-once area is in E4M8 where you cannot return to the start.  Such a place could contain some power ups to make sure you are in a good condition before the actual level starts but prevent you from saving them for later use.  In this case I wouldn't mind a stack of rockets.  In E4M8, you cannot leave the exit area but I in my opinion it is OK here even though I don't see that it adds anything to the level.

**Comment by Catoptromancy** [comment]
Very nice report! I have recently added Gamepad/touchscreen support as a milestone, would also cover keyboard-only players. Many issues you posted are listed under the Release Checklist. All your reports will help speed up the fixes.

**Comment by Xindage** [comment]
could this issue be closed and them reopened with just the ones present in the actual development version? this issue is getting confusing with mix of old and new bugs. (just collect what is present here, open a new issue and i'll close this one)

**Comment by andh0g** [comment]
Sound like a good idea to me!

Is there a better way to keep track of ideas and issues than this tracker?  It is not reasonable to open a ticker for each individual detail and collecting several, like in this ticket, means that some get fixed while others are still pending.  I'm not sure what it would be instead, maybe something more like a wiki.

**Comment by Xindage** [comment]
separate each issue by episode since we have 4 wqith 9 maps each i can fix them based on episodes one by one.

**Comment by mc776** [comment]
Let's never, ever post a giant mega-issue like this again. Or at least let's not have people replying to it with more issues instead of posting their own.

I've gone through this thread and checked what's left. Things that I can confirm are still outstanding are in **bold**.

From the OP:

E1M7
Sector 95 fixed.

E2M1
**Line 378 or 268 right next to it? I haven't tried in vanilla but I can see why that transparent vine texture would bleed.**
Backpack is fixed.

E2M2
This map has since been replaced and I don't know which one is the one this report is about, if it's still in Freedoom at all. The entire report is obsolete.

E2M3
There aren't that many sectors in the current map, so this is obsolete.

E2M4
Line 722 is a one-sided line now, and E2M4 was also part of the #788 move. Obsolete.

E2M5
See E2M4.

E2M6
T115 is a serpentipede, also #788.

E2M8
**I really like the end text line as it is, but having two cybers *really* ramps up the difficulty. Perhaps the player can be given a polaric and more cells in easy right away?**

E3M2
Obsoleted by #802.

E3M3
Xindage fixed these back in September, except point 2 but those are just monster closets not actual secrets.

E3M4, E3M5, E3M7
Xindage September.

E3M6
1. **Line 736 is an illusory wall. The stuff behind it should also be marked as not visible on automap, unless we change that to a door or something.**
2. There's no lava around sector 120.
3. Fixed.
4. Whatever this was about, it's gone now, or at least the line and sector references are obsolete.

E4M1
Line 801 is 1-sided and 1753 is from a sector that surrounds the yellow key. I don't trust the rest of these numbers to not be obsolete either.
The two nukage areas are physically different enough to justify it.
No need for sound blocks on a simple monster closet. Was this already fixed?

E4M2
OK to just leave as is - you'd know *something* is there but not how to reach it.

E4M9
Monster closet, not secret.

E4M6
**Sector 58 needs some alignment - the ceiling height might need some reconsideration too.**
Monster closet, not secret.
**Line 2343 texture is also misaligned.**

E4M7
1619 no need, in fact doing so would make the secret *too* obscure.
Others monster closet, I'm tempted to use an Inigo Montoya meme.

E4M8
I'm not even going to look at those l—**holy shit those actually are secrets!** (Northeast corner of that disconnected vaguely Mjolnir-shaped thing)
Adding a SKAG can seriously rebalance a map, depending on how easy it is to obtain one in E4M7, but if it's available in the episode *at all* **it is kinda weird to deprive the player on the final map**.


From perryprog's reply:

E1M6
E2M1
E3M6
E4M1
Can't find them and presumably fixed.

E3M7
Other stuff explicitly fixed.
**Line 448 is associated with a sector for stop-crush but that sectoro never *starts* crushing. And, frankly, shouldn't.**

E4M6
**Line 1465 issue is way the fuck above my pay grade.** Hawkwind posted a fix for it in the thread, but there have been other changes and fixes to this map since then so it will need to be re-done. (I believe the issue is the one big sector the trigger is inside - the 1-sided linedefs are all 1000-something compared with the 2k+ for the others.)


E4M7
Broken teleporters happen on maps all the time. Might be improved later but not a pressing bug.


Re BailShamber's issue: **what the hell even is going on there.** I'm looking at the map in SLADE and that setup is just bizarre-looking.


Moving on to andh0g's reply...

E3M6
Freedoom has way too goddamn many of these, but yes **we should fix sector 62.**

E3M7
I **think** we've gotten rid of this trilobite in the outer darkness (10880,24608)?

E4M6
It took me zero additional effort to hit this switch when approaching from an oblique angle, which makes diegetic sense anyway (no one's going to stand on the opposite side of the column and try to reach around it). No need to fix this.

E4M8
**The little square of control sectors in the south should be marked not-on-map.**

The rest of this post is a bunch of other thoughts and preferences and not clear, specific glitches to fix, and will be brought up again (or edited by a contributor at whim) for specific maps in the future if they really are problems.

**Comment by mc776** [comment]
Here is the above list with only the outstanding issues.

If you want I can repost this as a new issue and this one can be closed - and if I do that you have my advance express consent that if I disappear or fail to update the OP as things get fixed, any maintainer may close my issue.

E1M7
In the northern part of the room with the barred switch it is possible in vanilla to get stuck inside the instant-lift-crush terminal things - they don't kill you, you can't move, softlocked.

E2M1
Line 268 transparent vine texture looks like it would bleed.

E2M8
Easy mode gives only one tripod, making it inconsistent with the end text. Some additional health, armour and a more easily accessible polaric would help make a pair of tripods "easy" again.

E3M6
Line 736 is an illusory wall. The stuff behind it should also be marked as not visible on automap, unless we change that to a door or something.

E3M7
Line 448 is associated with a sector for stop-crush but that sectoro never *starts* crushing. And, frankly, shouldn't.
Double-check to make sure there's no trilobite in the outer darkness (last seen around (10880,24608))

E4M6
Sector 58 needs some alignment - the ceiling height might need some reconsideration too.
Line 2343 texture is also misaligned.
Sector 502 (thing that lowers into the blood to the southeast) has a 1-sided line as its lowest numbered linedef, resulting in garbage data for its "change tex+type" special. **high priority - may crash vanilla!**

E4M8
Chain of secrets in the Northeast corner of that disconnected vaguely Mjolnir-shaped thing need to be marked secret to prevent easy automap catching.
This map has no SKAG anywhere, which could result in an unduly large difficulty gap between playing full campaign and pistol start.
The little square of control sectors in the south should be marked not-on-map.
